### PR TITLE
Add a localeModels.rst subsubsection about numa performance.

### PR DESCRIPTION
**Pull request by gbtitus**

### CODE REVIEW

**Comment by gbtitus** [comment]
@ronawho , take a look?

File: doc/rst/technotes/localeModels.rst [hunk 1]
```diff
@@ -93,6 +93,62 @@ To use the NUMA locale model:
...
+locale model by adding the ability to split array data storage into
+blocks and distribute those blocks across NUMA domains.  Based on the
+internal term for array data storage, such arrays are called
+*multi-ddata* arrays.  Although only large arrays (2 MiB or more) can
+have multiple data blocks, the array addressing code to support them is
+always present when the NUMA locale model is used.  This code has turned
+out to cause large performance degradations in some cases, but it has
+also improved performance quite a bit in others.
+
+On the downside, array access in the NUMA locale model in 1.15 is much
```
Comment (ronawho) [inline doc/rst/technotes/localeModels.rst]: maybe "array indexing" instead of "array access"

File: doc/rst/technotes/localeModels.rst [hunk 2]
```diff
@@ -93,6 +93,62 @@ To use the NUMA locale model:
...
+
+.. code-block:: chapel
+
+    var R = 0..n;
+    var D = {R};
+    var A: [D] int;
+
+    forall i in D do ... A[i] ...;  // slower than 1.14
+    forall i in R do ... A[i] ...;  // slower than 1.14
+    forall a in A do ... a ...;     // same performance as 1.14
```
Comment (ronawho) [inline doc/rst/technotes/localeModels.rst]: same or better performance?

File: doc/rst/technotes/localeModels.rst [hunk 3]
```diff
@@ -93,6 +93,62 @@ To use the NUMA locale model:
...
+serial iteration over a 1-dimensional array (as opposed to its domain or
+range) remains as fast in 1.15 as it was in 1.14.
+
+Counteracting this to some extent, on Cray XE and XC systems with
+``CHPL_COMM=ugni`` and a hugepage-resident heap, the heap itself and any
+multi-ddata array will have proper NUMA locality, potentially improving
+performance.  As an example, as of 1.15, on a Cray XC system, the
+stream-ep benchmark with the NUMA locale model and ``CHPL_COMM=ugni``
+performs nearly the same as it does with the flat locale model,
+``CHPL_COMM=gasnet``, and ``CHPL_COMM_SUBSTRATE=mpi``, which has been
```
Comment (ronawho) [inline doc/rst/technotes/localeModels.rst]: I think we should avoid talking about gasnet-mpi here, and say something more along the lines of "achieves parity with the reference" or something.

File: doc/rst/technotes/localeModels.rst (hunk 1 above)
Comment (gbtitus) [inline doc/rst/technotes/localeModels.rst]: Will do.

File: doc/rst/technotes/localeModels.rst (hunk 3 above)
Comment (gbtitus) [inline doc/rst/technotes/localeModels.rst]: Okay, that's fair; will do.

**Comment by ronawho** [comment]
👍

**Pull request merged**